src/components/search/index.tsx

```diff
@@ -39,24 +39,22 @@ const HitsWrapper = styled.div`
 `;
 
 const indexName = config.header.search.indexName;
+const searchClient = algoliasearch(
+  config.header.search.algoliaAppId,
+  config.header.search.algoliaSearchKey
+);
+
+const Results = connectStateResults(({ searchState: state, searchResults: res, children }: any) =>
+  res && res.nbHits > 0 ? (
+    children
+  ) : (
+    <div className="no-results">No results for '{state.query}'</div>
+  )
+);
```

Comment:
This React component was also unnecessarily re-created on every render of the Search component.